### PR TITLE
Added reporting for configuration used when executing `getmyancestors…

### PR DESCRIPTION
**Pull request by changeling**
….py`.

Added reporting for configuration used when executing `getmyancestors.py`.
Added `.vscode/` and `scratchpad.py` to .gitignore.

### CODE REVIEW

**Comment by Linekio** [comment]
Hello, what is the redact_password variable for?

**Comment by changeling** [comment]
If `True`, `redact_password` causes the password to be replaced with `******`. For now, I left it hard-coded as a precaution, but it could easily be added to argparse to allow the user to include their password in the settings output.

Here's the difference between `True` and `False` values as examples:

```
time stamp:                                                               18:09:39 03/11/19 CDT
-u FamilySearch username                                                  my-username
-p FamilySearch password                                                  ******
-i List of individual FamilySearch IDs for whom to retrieve ancestors     None
-a Number of generations to ascend [4]                                    2
-d Number of generations to descend [0]                                   2
-m Add spouses and couples information [False]                            True
-r Add list of contributors in notes [False]                              True
-c Add LDS ordinances (need LDS account) [False]                          False
-v Increase output verbosity [False]                                      False
-t Timeout in seconds [60]                                                60
-o output GEDCOM file [stdout]                                            test.ged
-l output log file [stderr]                                               test.log
```

versus

```
time stamp:                                                               18:09:39 03/11/19 CDT
-u FamilySearch username                                                  my-username
-p FamilySearch password                                                  my-password
-i List of individual FamilySearch IDs for whom to retrieve ancestors     None
-a Number of generations to ascend [4]                                    2
-d Number of generations to descend [0]                                   2
-m Add spouses and couples information [False]                            True
-r Add list of contributors in notes [False]                              True
-c Add LDS ordinances (need LDS account) [False]                          False
-v Increase output verbosity [False]                                      False
-t Timeout in seconds [60]                                                60
-o output GEDCOM file [stdout]                                            test.ged
-l output log file [stderr]                                               test.log
```

**Comment by Linekio** [comment]
Ok, I understand the precaution.
But I think this isn't the purpose of this feature to keep the password.
Also I think it's better to write it into the gedcom, like this :
```
0 HEAD
1 CHAR UTF-8
1 GEDC
2 VERS 5.5.1
2 FORM LINEAGE-LINKED
1 SOUR getmyancestors
2 VERS 1.0
2 NAME getmyancestors
1 DATE 12 Mar 2019
2 TIME 14:35:11
1 NOTE @N1@
[............]
0 @N1@ NOTE Parameters
1 CONT -u FamilySearch username                                                  my-username
1 CONT -i List of individual FamilySearch IDs for whom to retrieve ancestors     None
1 CONT -a Number of generations to ascend [4]                                    2
1 CONT -d Number of generations to descend [0]                                   2
1 CONT -m Add spouses and couples information [False]                            True
1 CONT -r Add list of contributors in notes [False]                              True
1 CONT -c Add LDS ordinances (need LDS account) [False]                          False
1 CONT -v Increase output verbosity [False]                                      False
1 CONT -t Timeout in seconds [60]                                                60
1 CONT -o output GEDCOM file [stdout]                                            test.ged
1 CONT -l output log file [stderr]                                               test.log

```
I removed "timestamp" since it's a duplicate with DATE and TIME.

Advantages are :
- Possibility to adapt more easily to the CLI version
- Possibilite to keep its settings after import into a specific software.

@FlominatorTM  what is your opinion ?

**Comment by changeling** [comment]
I'd suggest that writing into the gedcom itself might create issues down the road. Just an intuitive feeling, with the profusion of programs out there. Personally, I think avoiding additional complications to the gedcom output is generally good.

That said, perhaps having the option to either put it in the gedcom, save as a `.settings` file for folks who want them tracked during a project, or allowing both, would be a reasonable approach.

The `.settings` file also offers the possibility of taking that in as CLI input to replicate a run, which could be useful for some folks, as would allowing them to edit the `.settings` file, perhaps to script batch runs, in which case the username and password might be useful. In that case, `******` would trigger a password request which they could include in their external script, or if they're unconcerned with security, the actual password in the `.settings` file would just work. If used in a project that involves tracking for reproducibility, as in academic research, the time stamp would also be handy.

If encoded in the gedcom, I agree the timestamp would be somewhat redundant, though grabbing the single NOTE structure, including timestamp, would be handy.


Regarding the advantages mentioned:

* In what ways would the gedcom encoding adapt to the CLI better than a settings file? Not saying it wouldn't. Just looking to understand.
* True. Though if that software is expecting the structure, it could also expect a `.settings` file.


TL;DR There are multiple use-cases possible which could benefit from a separate `.settings` file. Allowing the option might be worth considering, including the timestamp.

**Comment by Linekio** [comment]
Well, I think you're right.
Your code just needs somes changes:
- removing "print (settings_output)" line 947
- adding an argument for redact_password (default False)
- don't creating a * .settings file if the -o option is not specified. (Since the default output is sys.stdout, a "<stdout> .settings" file is created)
Could you make those changes?

**Comment by changeling** [comment]
Will do!

**Comment by changeling** [comment]
Changes made.

**Comment by Linekio** [comment]
Thanks for this changes.
Il'll change line 952 for "and not args.redact_password".

**Comment by changeling** [comment]
Just to clarify,

`if action.dest is 'p' and args.redact_password`

checks to see if the option we're testing for output is the `password` option. If it is, we then check to see if `--redact-password` is set to `True`. If that variable _is_ set to `True`, we replace the password in our output with `******`. If it is `False`, which is the default, we don't redact the password.

`if action.dest is 'p' and` **not** `args.redact_password`

disables the redaction functionality, so that we print the password regardless of the user's choice to redact it.

Could you remove the `not` from line 952, so that if the user chooses to redact, the output doesn't show the password?

**Comment by Linekio** [comment]
By default, I want the password not to be written, for security reasons.
So if "--redact-password" is not set, the password must be replaced with "******".
Perhaps the help message or the name of the option needs to be clarified.

**Comment by changeling** [comment]
Ah! I misunderstood your earlier comment. When you said default False for redact_password, I read that as "Don't redact password by default." To reverse the behavior, I think it would be best to change the option name to `--show-password`, then switch the code to have `password` and `******` in the opposite lines.

I'll open a PR for that.

**Comment by Linekio** [comment]
Ok so it's a ban understanding.
Maybe because my english is not very good ^^

**Comment by changeling** [comment]
> Ok so it's a ban understanding.
> Maybe because my english is not very good ^^

Ha! Your english seems fine. :) Redact isn't an everyday word.
